contrib/pkg/deprovision/gcp.go

```diff
@@ -0,0 +1,103 @@
+package deprovision
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/openshift/installer/pkg/destroy/gcp"
+	"github.com/openshift/installer/pkg/types"
+	typesgcp "github.com/openshift/installer/pkg/types/gcp"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+// gcpOptions is the set of options to deprovision a GCP cluster
+type gcpOptions struct {
+	logLevel  string
+	infraID   string
+	region    string
+	projectID string
+}
+
+// NewDeprovisionGCPCommand is the entrypoint to create the GCP deprovision subcommand
+func NewDeprovisionGCPCommand() *cobra.Command {
+	opt := &gcpOptions{}
+	cmd := &cobra.Command{
+		Use:   "gcp INFRAID --region=REGION --gcp-project-id=GCP_PROJECT_ID",
+		Short: "Deprovision GCP assets (as created by openshift-installer)",
+		Args:  cobra.ExactArgs(1),
+		Run: func(cmd *cobra.Command, args []string) {
+			if err := opt.Complete(cmd, args); err != nil {
+				log.WithError(err).Fatal("failed to complete options")
+			}
+			if err := opt.Validate(cmd); err != nil {
+				log.WithError(err).Fatal("validation failed")
+			}
+			if err := opt.Run(); err != nil {
+				log.WithError(err).Fatal("Runtime error")
+			}
+		},
+	}
+	flags := cmd.Flags()
+	flags.StringVar(&opt.logLevel, "loglevel", "info", "log level, one of: debug, info, warn, error, fatal, panic")
+	flags.StringVar(&opt.region, "region", "", "GCP region where the cluster is installed")
+	flags.StringVar(&opt.projectID, "gcp-project-id", "", "ID of the GCP project in which the cluster is installed")
+	return cmd
+}
+
+// Complete finishes parsing arguments for the command
+func (o *gcpOptions) Complete(cmd *cobra.Command, args []string) error {
+	o.infraID = args[0]
+	return nil
+}
+
+// Validate ensures that option values make sense
+func (o *gcpOptions) Validate(cmd *cobra.Command) error {
+	if o.region == "" {
+		cmd.Usage()
+		log.Info("Region is required")
+		return fmt.Errorf("missing region")
+	}
+	if o.projectID == "" {
+		cmd.Usage()
+		log.Info("GCP project ID is required")
+		return fmt.Errorf("missing project ID")
+	}
```

Comment:
I deliberated about this with myself. In the end, I went with the guidance from cobra that flags should be adjective-like (https://github.com/spf13/cobra#concepts). I don't have a strong distaste for required flags. If you do, then I can be persuaded to change it.

Note, however, that when creating a cluster on GCP, the `--gcp-project-id` flag is required. I don't think that it makes sense to have that as a positional argument that is only relevant and accepted when the cloud flag has a value of gcp. I like keeping the arguments the same between create-cluster and destroy--which is why the flag in destroy is named gcp-project-id instead of stutter-free project-id.

Reply:
One additional note is that we could grab the project ID from the creds file. It will not always be present, but if it is present, then the user-supplied project ID is redundant. I did not think it was worth the effort initially. I was going to mention it during standup today to see if it was something that we should pursue. If we did, then the gcp-project-id flag would be optional (although required if the creds did not have a project ID).

Reply:
Ok this is fine then.